flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -205,10 +221,11 @@ public SlotReport createSlotReport(ResourceID resourceId) {
 	// ---------------------------------------------------------------------
 
 	/**
-	 * Allocate the slot with the given index for the given job and allocation id. Returns true if
-	 * the slot could be allocated. Otherwise it returns false.
+	 * Allocate the slot with the given index for the given job and allocation id. If negative index is
+	 * given, a new auto increasing index will be generated. Returns true if the slot could be allocated.
```

Comment:
I think eventually we are going to remove `SlotID` and use `AllocationID` to identify the slots. But maybe not now, since we still need `SlotID` for the legacy RM.

Since we're not removing the `SlotID` at the moment, I'm in favor of keep the `SlotID` of dynamic slots valid, in other words each slot has a unique `SlotID`. Thus, we need to crate a unique id for each dynamic slot anyway, no matter positive or negative.